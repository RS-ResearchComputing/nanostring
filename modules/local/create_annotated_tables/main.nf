process CREATE_ANNOTATED_TABLES {
    tag "$sample_sheet"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "community.wave.seqera.io/library/r-dplyr_r-ggplot2_r-readr_r-stringr_r-tidyr:44c4e4fe69e11c2f"

    input:
    path counts
    path sample_sheet

    output:
    path "*ENDO.tsv"   , emit: annotated_endo_data
    path "*HK.tsv*"    , emit: annotated_hk_data
    path "*_mqc.tsv"   , emit: annotated_data_mqc
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    """
    write_out_prepared_gex.R $counts $sample_sheet

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(echo \$(R --version 2>&1) | sed 's/^.*R version //; s/ .*\$//')
        r-ggplot2: \$(Rscript -e "library(ggplot2); cat(as.character(packageVersion('ggplot')))")
        r-dplyr: \$(Rscript -e "library(dplyr); cat(as.character(packageVersion('dplyr')))")
        r-readr: \$(Rscript -e "library(readr); cat(as.character(packageVersion('readr')))")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    """
    touch ENDO.tsv
    touch HK.tsv
    touch mqc.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$(echo \$(R --version 2>&1) | sed 's/^.*R version //; s/ .*\$//')
        r-ggplot2: \$(Rscript -e "library(ggplot2); cat(as.character(packageVersion('ggplot')))")
        r-dplyr: \$(Rscript -e "library(dplyr); cat(as.character(packageVersion('dplyr')))")
        r-readr: \$(Rscript -e "library(readr); cat(as.character(packageVersion('readr')))")
    END_VERSIONS
    """
}
