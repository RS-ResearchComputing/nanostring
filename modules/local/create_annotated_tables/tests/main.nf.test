nextflow_process {

    name "Test Process CREATE_ANNOTATED_TABLES"
    script "../main.nf"
    process "CREATE_ANNOTATED_TABLES"
    tag "module"

    test("Should run without failures") {

        when {
            process {
                """
                // Normalized file
                input[0] = file(params.pipelines_testdata_base_path + 'nanostring/modules_test_data/test_profile/normalized_counts.tsv', checkIfExists: true)

                // Sample sheet
                input[1] = file(params.pipelines_testdata_base_path + 'nanostring/samplesheets/samplesheet_test.csv', checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
