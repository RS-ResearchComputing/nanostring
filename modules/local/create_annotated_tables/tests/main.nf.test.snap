{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "counts_Norm_GEX_ENDO.tsv:md5,5992886680cb6fc6f3e901e79752d544"
                ],
                "1": [
                    "counts_Norm_GEX_HK.tsv:md5,9d8a0503df864a82990648348131be21"
                ],
                "2": [
                    [
                        "Norm_GEX_ENDO_mqc.tsv:md5,5992886680cb6fc6f3e901e79752d544",
                        "Norm_GEX_HK_mqc.tsv:md5,9d8a0503df864a82990648348131be21"
                    ]
                ],
                "3": [
                    "versions.yml:md5,43b871def59fb08862ac85e636101fe5"
                ],
                "annotated_data_mqc": [
                    [
                        "Norm_GEX_ENDO_mqc.tsv:md5,5992886680cb6fc6f3e901e79752d544",
                        "Norm_GEX_HK_mqc.tsv:md5,9d8a0503df864a82990648348131be21"
                    ]
                ],
                "annotated_endo_data": [
                    "counts_Norm_GEX_ENDO.tsv:md5,5992886680cb6fc6f3e901e79752d544"
                ],
                "annotated_hk_data": [
                    "counts_Norm_GEX_HK.tsv:md5,9d8a0503df864a82990648348131be21"
                ],
                "versions": [
                    "versions.yml:md5,43b871def59fb08862ac85e636101fe5"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-20T14:10:50.997813905"
    }
}